_source/_posts/2023-07-27-spring-graphql-react.md

```diff
@@ -0,0 +1,1794 @@
+
+import com.okta.developer.demo.domain.Company;
+import com.okta.developer.demo.repository.CompanyRepository;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.graphql.GraphQlTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.graphql.test.tester.GraphQlTester;
+import reactor.core.publisher.Flux;
+
+import java.time.LocalDate;
+
+import static org.mockito.Mockito.when;
+
+@GraphQlTest(CompanyController.class)
+public class CompanyControllerTests {
+
+    @Autowired
+    private GraphQlTester graphQlTester;
+
+    @MockBean
+    private CompanyRepository companyRepository;
+
+    @Test
+    void shouldGetCompanies() {
+
+        when(this.companyRepository.findAll())
+                .thenReturn(Flux.just(new Company(
+                        "1234",
+                        "private",
+                        "12345678",
+                        "UK",
+                        LocalDate.of(2020, 1, 1),
+                        0,
+                        "Test Company",
+                        "active")));
+
+        this.graphQlTester
+                .documentName("companyList")
+                .variable("page", 0)
+                .execute()
+                .path("companyList")
+                .matchesJson("""
+                    [{
+                        "id": null,
+                        "SIC": "1234",
+                        "name": "Test Company",
+                        "status": "active",
+                        "category": "private",
+                        "companyNumber": "12345678",
+                        "countryOfOrigin": "UK"
+                    }]
+                """);
+    }
+}
+```
+
+Create the document file `companyList.graphql` containing the query definition for the test, in the folder `src/main/test/resources/graphql-test`:
+
+__companyList.graphql__
+```graphql
+query companyList($page: Int) {
+    companyList(page: $page) {
+        id
+        SIC
+        name
+        status
+        category
+        companyNumber
+        countryOfOrigin
+    }
+}
+```
+
+Update the test configuration in `build.gradle` file, so passed tests are logged:
+
+__build.gradle__
+```groovy
+tasks.named('test') {
+    useJUnitPlatform()
+
+    testLogging {
+        // set options for log level LIFECYCLE
+        events "failed", "passed"
+    }
+}
+```
+
+Run the test with:
+
+```shell
+./gradlew clean test
+```
+
+You should see logs for the successful tests:
+
+```
+...
+SpringBootGraphQlApiApplicationTests > contextLoads() PASSED
+
+CompanyControllerTests > shouldGetCompanies() PASSED
+...
+```
+
+
+### Add Neo4j seed data
+
+Let's add Neo4j migrations dependency for the seed data insertion. Edit the file `build.gradle` and add:
+
+__build.gradle__
+```groovy
+dependencies {
+    ...
+    implementation 'eu.michael-simons.neo4j:neo4j-migrations-spring-boot-starter:2.5.3'
```

Comment:
```suggestion
    implementation 'eu.michael-simons.neo4j:neo4j-migrations-spring-boot-starter:2.5.3'
    ...
```

Add an ellipsis after because there are more `testImplementation` dependencies, or remove the dependencies wrapper.